### PR TITLE
-Add ForgetPassword Screen

### PR DESCRIPTION
**Pull request by pariami**
Add ForgetPassword Screen
-Add ForgetPasswordContent Test
-Add empty Inject for ViewModels

### CODE REVIEW

File: app/src/main/java/com/training/app/trainingapp/main/view/forget_password/ForgetPasswordViewModel.kt [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+    }
+
+    fun onSubmitButtonClicked() {
+        validateEmail()
+        if (_emailValidateState.value) {
+            sendEmailForPasswordRecovery()
+        }
+    }
+
+    private fun sendEmailForPasswordRecovery() {
```
Comment (ms-dehghani) [inline app/src/main/java/com/training/app/trainingapp/main/view/forget_password/ForgetPasswordViewModel.kt:31]: I think we don need this in current state and we should add it in another issue

File: app/src/main/java/com/training/app/trainingapp/main/view/forget_password/ForgotPasswordScreen.kt [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+import androidx.compose.runtime.collectAsState
+
+@Composable
+fun ForgetPasswordScreen(viewModel: ForgetPasswordViewModel) {
+    ForgetPasswordContent(
+        viewModel.userEmail.collectAsState().value,
+        viewModel.emailValidateState.collectAsState().value,
+        {},
+        {})
+}
```
Comment (ms-dehghani) [inline app/src/main/java/com/training/app/trainingapp/main/view/forget_password/ForgotPasswordScreen.kt:13]: Please add preview function for this view

File: app/src/main/java/com/training/app/trainingapp/main/view_model/RegisterViewModel.kt
Comment (ms-dehghani) [inline app/src/main/java/com/training/app/trainingapp/main/view_model/RegisterViewModel.kt:1]: Please rollback this class.
changing this class is not related to this issue

Comment (pariami) [reply app/src/main/java/com/training/app/trainingapp/main/view_model/RegisterViewModel.kt:1]: In this case, the project could not be built and it gave an error that HiltViewmodel needs inject, that's why I added it.

File: app/src/main/java/com/training/app/trainingapp/main/view/forget_password/ForgetPasswordViewModel.kt (hunk 1 above)
Comment (pariami) [reply app/src/main/java/com/training/app/trainingapp/main/view/forget_password/ForgetPasswordViewModel.kt:31]: I think it can stay because no implementation has been done for it, it is only specified here what is going to be done when the button is clicked.

**Review by ms-dehghani — APPROVED** [review]

File: app/src/main/java/com/training/app/trainingapp/main/view/forget_password/ForgetPasswordViewModel.kt (hunk 1 above)
Comment (viamgr) [reply app/src/main/java/com/training/app/trainingapp/main/view/forget_password/ForgetPasswordViewModel.kt:31]: At least add a //todo comment if there is no implementation.

File: app/src/main/java/com/training/app/trainingapp/main/view/forget_password/ForgotPasswordScreen.kt [hunk 3]
```diff
@@ -0,0 +1,20 @@
+package com.training.app.trainingapp.main.view.forget_password
+
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.collectAsState
+import androidx.compose.ui.tooling.preview.Preview
+
+@Composable
+fun ForgetPasswordScreen(viewModel: ForgetPasswordViewModel) {
+    ForgetPasswordContent(
+        viewModel.userEmail.collectAsState().value,
```
Comment (viamgr) [inline app/src/main/java/com/training/app/trainingapp/main/view/forget_password/ForgotPasswordScreen.kt:10]: Add clarified names for the function arguments.

File: app/src/main/java/com/training/app/trainingapp/utils/TestTags.kt [hunk 4]
```diff
@@ -4,5 +4,8 @@ object TestTags {
 
     const val EMAIL_TEXT_FILED_ID = "emailTextFiledID"
     const val CHECK_BUTTON_ID = "checkButtonID"
+    const val PASSWORD_RECOVERY_BUTTON_ID = "PasswordRecoveryButtonID"
+    const val PASSWORD_RECOVERY_EMAIL_TEXT_FILED_ID = "passwordRecoveryEmailTextFiledID"
+
```
Comment (viamgr) [inline app/src/main/java/com/training/app/trainingapp/utils/TestTags.kt:9]: Remote this extra line

**Review by viamgr — APPROVED** [review]

**Review by ms-dehghani — APPROVED** [review]

**Review by viamgr — APPROVED** [review]
